pkg/refactor/events/events.go

```diff
@@ -0,0 +1,107 @@
+package events
+
+import (
+	"context"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+)
+
+// wraps the cloud-event and adds contextual information.
+type Event struct {
+	Event      *cloudevents.Event
```

Comment:
I'd strongly consider replacing this `*cloudevents.Event` with a generic `[]byte` field. I've had trouble with the `cloudevents` library in the past because it performs heavy validation of the data when unmarshaling. And it doesn't validate it as much for marshaling, so it can result in this struct being capable of storing information that can't be retrieved.

Reply:
I don't think you can avoid the problems of the cloudevents library by using the json package. The json package defers to custom marshal/unmarshal logic if the object satisfies a certain interface. 

You're right about the error just being shifted, but I thought it might be preferrable to keep this logic simple and let the caller decide if they want to handle this error. Handling it internally could make this package more complex. But if you want, we'll do that.